contracts/IStakingLocking.sol

```diff
@@ -1,33 +1,33 @@
 pragma solidity ^0.4.24;
 
-import "./ILockManager.sol";
-
 
 interface IStakingLocking {
-    event Locked(address indexed account, uint256 indexed lockId, uint256 amount, address manager, bytes data);
-    event Unlocked(address indexed account, uint256 indexed lockId, uint256 amount, address manager, bytes data);
-    event LockAmountChanged(address indexed account, uint256 indexed lockId, uint256 amount);
-    event LockManagerChanged(address indexed account, uint256 indexed lockId, address manager);
-    event LockDataChanged(address indexed account, uint256 indexed lockId, bytes data);
+    event Locked(address indexed account, address indexed lockManager, uint256 amount, uint256 allowance, bytes data);
+    event Unlocked(address indexed account, address indexed lockManager, uint256 amount, bytes data);
+    event LockAmountChanged(address indexed account, address indexed lockManager, uint256 amount, bool increase);
+    event LockAllowanceChanged(address indexed account, address indexed lockManager, uint256 allowance, bool increase);
+    event LockManagerChanged(address indexed account, address indexed oldLockManager, address newLockManager);
+    event LockDataChanged(address indexed account, address indexed lockManager, bytes data);
 
-    function lock(uint256 _amount, address _manager, bytes _data) external returns (uint256 _lockId);
-    function unlock(address _account, uint256 _lockId) external;
-    function decreaseLockAmount(address _account, uint256 _lockId, uint256 _newAmount) external;
-    function setLockManager(address _account, uint256 _lockId, ILockManager _newManager) external;
-    function setLockData(address _account, uint256 _lockId, bytes _newData) external;
-    function transfer(address _to, uint256 _toLockId, uint256 _amount) external;
-    function transferFromLock(address _account, uint256 _lockId, address _to, uint256 _toLockId, uint256 _amount) external;
+    function lock(uint256 _amount, address _lockManager, uint256 _allowance, bytes _data) external;
+    function unlock(address _account, address _lockManager) external;
+    function increaseLockAllowance(address _lockManager, uint256 _allowance) external;
+    function decreaseLockAllowance(address _account, address _lockManager, uint256 _allowance) external;
+    function increaseLockAmount(address _account, address _lockManager, uint256 _amount) external;
+    function decreaseLockAmount(address _account, address _lockManager, uint256 _amount) external;
+    function setLockManager(address _account, address _newLockManager) external;
+    function setLockData(address _account, bytes _newData) external;
```

Comment:
I think we should be able to remove this, we are only using it for the `TimeLockManager` which could be implemented in a different way.

Reply:
I understood offline that we should even remove data from lock storage, right?